ctl/filebuffer.go

```diff
@@ -0,0 +1,116 @@
+package ctl
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+func NewFileBuffer(max int, temp string) *FileBuffer {
+	return &FileBuffer{max: max, tempDir: temp}
+}
+
+type FileBuffer struct {
+	max     int
+	buf     bytes.Buffer
+	file    *os.File
+	tempDir string
+	reading bool
+	files   []*os.File
+	mu      sync.Mutex
+}
+
+func (fb *FileBuffer) Write(p []byte) (n int, err error) {
+	if fb.reading {
```

Comment:
I noticed a mutex covering some members of the FileBuffer. Is there no risk of concurrent access in the Write method?

Reply:
not in how im using it, basically you do single process writing to fill it up and then launch potentially multiple readers